index.js

```diff
@@ -68,70 +65,74 @@ let pads = {
         data.results = [];
 
         rs.forEach(function (value) {
-            entrySet = {padName: value, lastEdited: "", userCount: 0};
+            entrySet = { padName: value, lastEdited: '', userCount: 0 };
             data.results.push(entrySet);
         });
 
+        let getEdited = [];
         if (data.results.length > 0) {
             data.results.forEach(function (value) {
-                let resultObject = api.getLastEdited(value.padName);
-                value.lastEdited = resultObject.lastEdited;
-                resultObject = api.padUsersCount(value.padName);
-                value.userCount = resultObject.padUsersCount;
+                getEdited.push(
+                    api.getLastEdited(value.padName)
```

Comment:
Fix was here . Need to actually wait for promises to resolve before we return the data or that data won't get passed back to the view.